### PR TITLE
tls: reduce the security strength of the CCM8 cipher suites

### LINKED ISSUE #16154: Reconsider strength of CCM_8 TLS ciphers
As far as I can tell, we don't give any particular consideration to the truncated authentication tag of the CCM_8 ciphers when we're considering cipher strength and security level.  E.g., at `openssl ciphers -s -v 'ALL@SECLEVEL=2' we still see ECDHE-ECDSA-AES256-CCM8, AES128-CCM8, etc., and I think that TLS_AES_128_CCM_8_SHA256 is also treated as being strong.
It's not entirely clear that we should be treating ciphers that only have a 64-bit authentication tag as providing that level of security.

(Inspired by https://github.com/openssl/openssl/issues/13689#issuecomment-752220290)

### PR DESCRIPTION
**Pull request by paulidale**
CCM8 suites have a 64 bit authentication tag which is pretty short.

The changes reduces these cipher suites to security levels below 2.  The exact level chosen is open to discussion.

Fixes #16154 

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: doc/man3/SSL_CTX_set_security_level.pod [hunk 1]
```diff
@@ -83,8 +83,9 @@ using MD5 for the MAC is also prohibited.
 
 Security level set to 112 bits of security. As a result RSA, DSA and DH keys
 shorter than 2048 bits and ECC keys shorter than 224 bits are prohibited.
-In addition to the level 1 exclusions any cipher suite using RC4 is also
-prohibited. SSL version 3 is also not allowed. Compression is disabled.
+In addition to the level 1 exclusions any cipher suite using RC4 and any
+cipher suite using CCM with a 64 authentication tag are also prohibited.
```
Comment (slontis) [inline doc/man3/SSL_CTX_set_security_level.pod:87]: 64 bit

Comment (paulidale) [reply doc/man3/SSL_CTX_set_security_level.pod:87]: I'm sure I typed that :(

Fixed now.

**Comment by mattcaswell** [comment]
OTC: We will consider this post 3.0

**Comment by paulidale** [comment]
Was kind of hoping for a 3.0.1/3.1 decisions.  Master feels fine for this change.

**Comment by kaduk** [comment]
This seems like it has too much risk of application breakage to be appropriate for 3.0.1, to me.

**Comment by mattcaswell** [comment]
OTC: Reducing the strength of CCM8 is ok for master. We should consider whether a data driven solution would be better.

**Comment by paulidale** [comment]
unrequited.

**Pull request closed without merging**